ando/engine.py

```diff
@@ -36,16 +36,35 @@ def _test_is_included(my_list_of_lists, list_elem):
 
         return False
 
-    new_list_of_lists = []
+    def _merge_duplicates(my_list_of_lists, max_length=3):
```

Comment:
1. a docstring is missing

2. this does not look super efficient ;) ... can you add something in the docstring saying that this might have to be re-implemented more efficiently in the future?